types/ember-data/test/promises.ts

```diff
@@ -18,3 +18,7 @@ const promiseFind = store.findRecord('person', 1);
 promiseFind.content; // $ExpectType Person | undefined
 
 promiseFind.get('firstName');
+
+let promiseArray = DS.PromiseArray.create({ promise: RSVP.Promise.resolve([1]) });
+let promiseObject = DS.PromiseObject.create({ promise: RSVP.Promise.resolve({ value: 1 }) });
+let promiseManyArray = DS.PromiseManyArray.create({ promise: store.findAll('person') });
```

Comment:
Can you add a type assertion on `promiseManyArray.promise` to make sure it's a promise that resolves to an `ArrayProxy` and not an `Ember.Array`? e.g.
```ts
import ArrayProxy from '@ember/array/proxy';
⋮
assertType<RSVP.Promise<ArrayProxy<DS.Model>>>(promiseManyArray.promise);
```
This will make sure we don't lose that now that we need to pass it in to `PromiseArray`.